### PR TITLE
[jsk_perception] Replace deprecated nodes/nodelets to new node/nodelets

### PR DESCRIPTION
**Pull request by iory**
For https://github.com/jsk-ros-pkg/jsk_recognition/issues/1724 and https://github.com/jsk-ros-pkg/jsk_recognition/pull/1696

### CODE REVIEW

File: jsk_perception/jsk_perception_nodelets.xml [hunk 1]
```diff
@@ -174,7 +174,13 @@
       compute superpixels
     </description>
   </class>
-  <class name="jsk_perception/HoughCircleDetector" type="jsk_perception::HoughCircleDetector"
+  <class name="opencv_apps/hough_lines" type="hough_lines::HoughLinesNodelet"
```
Comment (wkentaro) [inline jsk_perception/jsk_perception_nodelets.xml]: Maybe the nodelet name should not be changed.

File: jsk_perception/CMakeLists.txt [hunk 2]
```diff
@@ -139,12 +138,13 @@ jsk_perception_nodelet(src/bounding_box_to_rect.cpp "jsk_perception/BoundingBoxT
 jsk_perception_nodelet(src/rect_array_actual_size_filter.cpp "jsk_perception/RectArrayActualSizeFilter" "rect_array_actual_size_filter")
 jsk_perception_nodelet(src/polygon_array_color_likelihood.cpp "jsk_perception/PolygonArrayColorLikelihood" "polygon_array_color_likelihood")
 jsk_perception_nodelet(src/polygon_array_color_histogram.cpp "jsk_perception/PolygonArrayColorHistogram" "polygon_array_color_histogram")
-jsk_perception_nodelet(src/edge_detector.cpp "jsk_perception/EdgeDetector" "edge_detector")
+jsk_perception_nodelet(src/edge_detector.cpp "edge_detection/edge_detection" "edge_detector")
```
Comment (wkentaro) [inline jsk_perception/CMakeLists.txt]: Maybe this should be as below:
```
jsk_perception_nodelets(${opencv_apps_SOURCE_DIR}/src/nodelets/edge_detector.cpp "jsk_perception/EdgeDetector" "edge_detector")
```

Comment (wkentaro) [inline jsk_perception/CMakeLists.txt]: Maybe the source file should be removed and this line also.

Comment (wkentaro) [inline jsk_perception/CMakeLists.txt]: Or does this work?

```
jsk_perception_nodelet(${opencv_apps_LIBRARIES} "edge_detection/edge_detection" "edge_detector")
```

Comment (iory) [inline jsk_perception/CMakeLists.txt]: It does not work.
Should I removed abobe lines?

Comment (wkentaro) [inline jsk_perception/CMakeLists.txt]: Maybe this does work:

https://github.com/iory/jsk_recognition/pull/1/files

**Comment by iory** [comment]
Should I remove following lines?
https://github.com/jsk-ros-pkg/jsk_recognition/blob/master/jsk_perception/jsk_perception_nodelets.xml#L189-L194

**Comment by wkentaro** [comment]
Maybe you can do like this.

https://github.com/jsk-ros-pkg/jsk_recognition/blob/master/jsk_pcl_ros/jsk_pcl_nodelets.xml#L13

2016年6月4日土曜日、iory<notifications@github.com>さんは書きました:

> Should I remove following lines?
>
> https://github.com/jsk-ros-pkg/jsk_recognition/blob/master/jsk_perception/jsk_perception_nodelets.xml#L189-L194
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/jsk-ros-pkg/jsk_recognition/pull/1726#issuecomment-223747295>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AEHFk-mnhTfNlmtVMGfwFVDtpbMzCjdKks5qIUuMgaJpZM4Ithar>
> .
>


-- 
和田 健太郎 / Kentaro Wada
http://wkentaro.com

**Comment by iory** [comment]
hydro doesn't support opencv_apps.
sometimes, this repository(jsk_recognition) will cease to support hydro.
don't merge yet.

**Comment by iory** [comment]
@k-okada 
Please merge this.

**Comment by wkentaro** [comment]
Looks awesome. :sparkles:

**Comment by k-okada** [comment]
is it possible to specify version tag at https://github.com/jsk-ros-pkg/jsk_recognition/pull/1726/commits/22ca78b576207098ca24bae47f0cf4d2a273f8a0 ? 
and can we set the version number which does not require .travis_before_script_opencv_apps.bash ?
because I think we're almost end of hydro support.

**Comment by wkentaro** [comment]
@iory Please rebase origin/master.

**Comment by iory** [comment]
rebased

**Comment by wkentaro** [comment]
+1 to me. @k-okada what do you think?

**Comment by k-okada** [comment]
opencv_apss is now compilable on hydro, so please check again

**Comment by iory** [comment]
OK, I will re-test it.

**Comment by iory** [comment]
test passed.
Please check this PR.

**Comment by k-okada** [comment]
one last question, what happens if we merge https://github.com/ros-perception/opencv_apps/pull/21. Do we need to change this PR?

**Comment by iory** [comment]
Even if the ros-perception/opencv_apps#21 merged, we can successfully call edge_detector.
so please merge ros-perception/opencv_apps#21 and this PR.

**Comment by k-okada** [comment]
#21 is api break, so I'll merge after next release

--
◉ Kei Okada

**Comment by iory** [comment]
OK, I am waiting for next release.

**Comment by k-okada** [comment]
so do you men if we do not merge https://github.com/ros-perception/opencv_apps/pull/21, this pr does not work, is it correct?

**Comment by iory** [comment]
This PR work. If this PR merged, jsk_perception's edge_detector will be deprecated.
and if we call jsk_perception's edge_detector, opencv_apps's edge_detector will call.
This effect enable, whether  ros-perception/opencv_apps#21merged or not.

**Comment by k-okada** [comment]
closed via #1859

**Pull request closed without merging**